seed/challenges/01-responsive-web-design/applied-visual-design.json

```diff
@@ -2252,7 +2252,7 @@
         "<div id=\"ball\"></div>"
       ],
       "tests": [
-        "assert(code.match(/opacity:\\s*?0?\\.1;/gi), 'message: The <code>keyframes</code> rule for fade should set the <code>opacity</code> property to 0.1 at 50%.');"
+        "assert(code.match(/@keyframes fade\\s*?{\\s*?50%\\s*?{\\s*?left:\\s*?60%;\\s*?opacity:\\s*?0?\\.1;/gi), 'message: The <code>keyframes</code> rule for fade should set the <code>opacity</code> property to 0.1 at 50%.');"
       ],
```

Comment:
There's one downside to these regex tests - that they can be very strict about the order the camper has to put the rules. For example, if I ignore the generous whitespace provided in the seed code 😄 and put the new `opacity` rule before the `left` one in there, the challenge won't pass. I think we can tweak this regex a bit to make it a bit more flexible, and check the camper didn't delete the rule already there. 

Since there are only two (right) ways of where to put the new rule (before or after the existing `left: 60%;`), we can tweak this regex to allow both ways. One way is to change this part of the regex:

`left:\\s*?60%\\s*?opacity:\\s*?0?\\.1;`

To use something like a non-capturing alternating group:

`(?:left:\\s*?60%;\\s*?opacity:\\s*?0?\\.1;|opacity:\\s*?0?\\.1;\\s*?left:\\s*?60%;)`

And there are certainly other ways to accomplish this, if you know a more elegant solution, use that 👍  (Full disclosure, I only checked this on an online regex site, not in the seed code, it may need further refinement).